### ISSUE TITLE
TypeError: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by onehorsetown** [Microsoft/TypeScript#27214]
**TypeScript Version:**  3.1.0-dev.20180919

**Search Terms:**
Cannot read property 'kind' of undefined

**Code**

```ts
enum Color {
  RED,
  BLUE
}

class Thing {
  public color: Color;

  public constructor() {
    this.color = Color.RED;
  }
}
```

**Expected behavior:**
Compiles fine in 3.0.3

**Actual behavior:**
tsc --strict bug.ts -> crashes compiler

Only with --strict (specifically --strictNullChecks AND --strictPropertyInitialization)

```
TypeError: Cannot read property 'kind' of undefined
at isConstraintPosition
at getConstraintForLocation
at getInitialOrAssignedType
at getTypeAtFlowAssignment
...
```

**Playground Link:** 
https://bit.ly/2MJaboV

Does not crash though.   Compiles fine with TS 3.0.3

**Related Issues:** 
#27092
#26153

**Comment by onehorsetown** [comment]
PR fix for #26433 does not fix this issue.

**Comment by j-oliveras** [comment]
Duplicate of #26978.

Trick: to search for crashes is better search by top callstack function (in this case `isConstraintPosition`).

**Comment by j-oliveras** [comment]
@weswigham this is fixed by your #27246 (tested with nigthly 3.1.0-dev.20180921).

**Comment by onehorsetown** [comment]
Confirmed fixed in 3.1.0-dev.20180921
